### ISSUE TITLE
Bug: CLN Peer connection not working

### ISSUE THREAD

**Issue by SeverinAlexB** [jamaljsr/polar#747]
**Describe the bug**

I created a new network with 3 CLN nodes. They are not able to establish peer connections with each other. Internal IP address is wrong.

**To Reproduce**
1. Create new network from scratch with a bitcoin node and 2 cln nodes.
2. Start the network
3. See logs or try `lightning-cli connect` with the given "P2P internal" address.

**Expected behavior**

Peer connection should be established on network start and should be possible with the "P2P Internal" address.

**Screenshots**

Looks like all of the nodes in the network get the same (wrong) internal IP address.

![CleanShot 2023-07-24 at 11 02 00@2x](https://github.com/jamaljsr/polar/assets/8782386/b713c0b2-ad1d-47f9-896a-b26f7a6f4210)
![CleanShot 2023-07-24 at 11 02 06@2x](https://github.com/jamaljsr/polar/assets/8782386/69336553-cc2d-4a2d-8e23-a9cfcb87250c)
![CleanShot 2023-07-24 at 11 02 12@2x](https://github.com/jamaljsr/polar/assets/8782386/6d4f7949-13aa-4096-beeb-85e61c9c7d55)

**Desktop (please complete the following information):**
- OS: macOS 13.2
- Polar Version: 2.0.0
- Docker Version: 24.0.2, build cb74dfc
- Docker Compose Version: 2.19.1

**Workaround**

Use the actual docker ip addresses and establish peer connection manually.

- Figure out the docker network name `docker network ls`. Example: `polar-network-3_default`.
- Lookup the container ip addresses with `docker network inspect %networkName%`.
- Manually connect with the actual IP address `lightning-cli connect`.

**Additional context**

CLN logs:
```
2023-07-24T03:57:27.238Z UNUSUAL plugin-plugin.js: {\"type\":\"lightning\",\"name\":\"LightningError\",\"message\":\"All addresses failed: 0.0.0.0:9735: Cryptographic handshake: peer closed connection (wrong key?). \",\"code\":401,\"fullType\":\"lightning\"}
2023-07-24T03:57:29.469Z DEBUG   gossipd: seeker: no peers, waiting
```

Docker network command for the actual IP address:

```
$ docker network inspect polar-network-3_default
[
    {
        "Name": "polar-network-3_default",
        "Id": "937387c872a4965dd6c4a81ab6552e034e49b114a0538880ca4e8cbc62a1ddf9",
        "Created": "2023-07-24T04:23:49.488860846Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.23.0.0/16",
                    "Gateway": "172.23.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "16084e982bc268dcc76afd2d8ae18c7b72a6ecc7fde7c2cec4a26dd8e0b2f3d9": {
                "Name": "polar-n3-bob",
                "EndpointID": "48554db7917fa4fc4da38afb72bd9dd67905c5da2b23f7fc872398061739aa2b",
                "MacAddress": "02:42:ac:17:00:03",
                "IPv4Address": "172.23.0.3/16",
                "IPv6Address": ""
            },
            "cc526ee7b8b3152ea0e1df123badbd67aa672864b88eeafb20fa2027e0519924": {
                "Name": "polar-n3-backend1",
                "EndpointID": "eec21554729a3d665d6cd5f548e5ca3c6e4f1ba65f602b0851b08d22a50536dd",
                "MacAddress": "02:42:ac:17:00:05",
                "IPv4Address": "172.23.0.5/16",
                "IPv6Address": ""
            },
            "e6fe147dc2926030db48375b49d68e0a5c1915623079b31fa4b11c3052efacc9": {
                "Name": "polar-n3-alice",
                "EndpointID": "aee1edaf5255b34bf279e9eea48140cad9845417d7f75ad2984056e0c371ebc6",
                "MacAddress": "02:42:ac:17:00:02",
                "IPv4Address": "172.23.0.2/16",
                "IPv6Address": ""
            },
            "ee1801605cd2ec4c89b5d63b2e8633b90b1d2facc038f02c523531040b713299": {
                "Name": "polar-n3-carol",
                "EndpointID": "dd6e7ad53af99c956a8f7ab581f39eb5fc9533d2f4c32097919fbbc88d2f481b",
                "MacAddress": "02:42:ac:17:00:04",
                "IPv4Address": "172.23.0.4/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "default",
            "com.docker.compose.project": "polar-network-3",
            "com.docker.compose.version": "2.19.1"
        }
    }
]
```

Connection works with actual docker ip.
```
lightning-cli connect 02af90cf28f4aa6f211695a9376128a4ca9311aef226927fffcbb1538b2b507931@172.23.0.3:9735
{
   "id": "02af90cf28f4aa6f211695a9376128a4ca9311aef226927fffcbb1538b2b507931",
   "features": "08a0000a0269a2",
   "direction": "out",
   "address": {
      "type": "ipv4",
      "address": "172.23.0.3",
      "port": 9735
   }
}
```

**Comment by jamaljsr** [comment]
Great bug report 👌. Thank you for all of the details. You are correct that the `0.0.0.0` IP is invalid and shouldn't be displayed.

Thanks for the suggestion on a solution. I think there's a simpler approach. Instead of looking up the internal IP address via the Docker APIs, we can simply use the internal hostname of the node (ex: `bob`). This will always resolve to the correct IP address from any other node in the network.

I've confirmed this works as expected when connecting `alice` to `bob`:
```sh
clightning@alice:/$ lightning-cli connect 0244b990bb066bf9a10470821c90d24dab499a538be219824f918755fd2357b500@0.0.0.0:9735
{
   "code": 401,
   "message": "All addresses failed: 0.0.0.0:9735: Cryptographic handshake: peer closed connection (wrong key?). "
}
clightning@alice:/$ lightning-cli connect 0244b990bb066bf9a10470821c90d24dab499a538be219824f918755fd2357b500@bob:9735
{
   "id": "0244b990bb066bf9a10470821c90d24dab499a538be219824f918755fd2357b500",
   "features": "08a0000a0269a2",
   "direction": "out",
   "address": {
      "type": "ipv4",
      "address": "192.168.128.3",
      "port": 9735
   }
}
clightning@alice:/$ 
```

I'll work on implementing this fix. It should be a fairly low lift.

**Comment by SeverinAlexB** [comment]
Ahhh didn't think about that! Thanks for fixing 🙏

**Comment by bshramin** [comment]
@jamaljsr Any way I can help in fixing this?

**Comment by jamaljsr** [comment]
This is fixed in #762
